### PR TITLE
Add plugin: MagicCalendar

### PR DESCRIPTION
**Pull request by Vaccarini-Lorenzo**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:
https://github.com/Vaccarini-Lorenzo/iCalObsidianSync

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [ ] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
[html, body {](https://github.com/Vaccarini-Lorenzo/iCalObsidianSync/blob/aecdd45f0c200792459b3e0570023a3bce9c0073/styles.css#L1-L4)
Don't modify core Obsidian styling, only add styles relevant to your plugin.

[.content {](https://github.com/Vaccarini-Lorenzo/iCalObsidianSync/blob/aecdd45f0c200792459b3e0570023a3bce9c0073/styles.css#L162)
Some of these class names are pretty generic, I would recommend making them more unique.

[const pluginPath =`${basePath}/.obsidian/plugins/ical-obsidian-sync`;](https://github.com/Vaccarini-Lorenzo/iCalObsidianSync/blob/aecdd45f0c200792459b3e0570023a3bce9c0073/src/plugin/main.ts#L85)
Users can configure where they place their settings files, use `this.plugin.manifest.dir` instead.

[const general = containerEl.createEl("h3", {text: "General"})](https://github.com/Vaccarini-Lorenzo/iCalObsidianSync/blob/aecdd45f0c200792459b3e0570023a3bce9c0073/src/plugin/appSetting.ts#L49)
[Don't use a heading for your general settings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[startDate: dayjs(from ?? dayjs().startOf("month")).format(this.dateFormat),](https://github.com/Vaccarini-Lorenzo/iCalObsidianSync/blob/aecdd45f0c200792459b3e0570023a3bce9c0073/src/iCloudJs/calendar.ts#L186-L187)
I would recommend to use the `moment` library instead, it's already bundled with obsidian `import {moment} from "obsidian"`



As for the CORS proxy, you can just use the `requestUrl` function from the Obsidian API, that will take care of CORS automatically.

**Comment by Vaccarini-Lorenzo** [comment]
Thanks for your feedback! 
The new release [1.1.1](https://github.com/Vaccarini-Lorenzo/iCalObsidianSync/releases/tag/1.1.1) should be compliant with what you suggested. Unfortunately, the only point I couldn't change was the one about the plugin path: ```this.plugin.manifest.dir``` outputs a relative path and the plugin needs an absolute path to the pattern files needed for the NLP module.

**Comment by joethei** [comment]
That was specifically about the `.obsidian` part of the path, users can change the name of that folder with a setting.
I would recommend to use that in combination with the basePath you already.

**Comment by Vaccarini-Lorenzo** [comment]
Oh, my bad. Fixed it
You can find everything in [1.1.3]((https://github.com/Vaccarini-Lorenzo/iCalObsidianSync/releases/tag/1.1.3) release

**Comment by Vaccarini-Lorenzo** [comment]
The project is currently going through a major refactor in order to extend its functionalities to other calendars too. For this reason, the repository name and the plugin id will soon change. Would it be a problem?

**Comment by liamcain** [comment]
> For this reason, the repository name and the plugin id will soon change. Would it be a problem?

@Vaccarini-Lorenzo  Yes, changing the plugin ID is quite tricky. If you plan on changing the plugin ID, do that now before I merge this. Obsidian uses the plugin ID for checking for updates and storing the plugin data, so once the plugin is merged, its not recommended to change the ID _ever_. Changing the plugin name, however, is fine. That's just for display purposes so it can be changed freely.

@ me once the plugin ID has been changed and I can merge this in. Thanks!

File: community-plugins.json [hunk 1]
```diff
@@ -8286,5 +8286,12 @@
     "author": "Ellpeck",
     "description": "Quickly and easily share individual notes online using an anonymized link. Also easy to self-host!",
     "repo": "Ellpeck/ObsidianJustSharePlease"
+  },
+  {
+    "id": "ical-obsidian-sync",
+    "name": "iCalSync",
+    "author": "Vaccarini Lorenzo",
+    "description": "Synchronize iCalendar with your notes",
```
Comment (liamcain) [inline community-plugins.json:8294]: Small nitpick but I don't think Apple brands Calendar as iCalendar or iCal.

**Comment by liamcain** [comment]
Hi @Vaccarini-Lorenzo, I just wanted to make sure you saw my less message regarding getting this merged. Code-wise this plugin looks good to me! Just left a couple small comments.

**Comment by Vaccarini-Lorenzo** [comment]
Hi @liamcain! Thanks for your message, I'll get back to you as soon as possible addressing what you pointed out. Unfortunately, the development of this plugin has been briefly paused but will resume shortly. Thank you!

**Comment by Vaccarini-Lorenzo** [comment]
Hi @liamcain, the plugin ID has been changed! You can find everything [here](https://github.com/Vaccarini-Lorenzo/MagicCalendar/releases/tag/1.1.6).
Thank you for your time!

**Comment by liamcain** [comment]
Hi @Vaccarini-Lorenzo You'll also need to update the plugin ID and plugin name here in the pull request. Thanks!

**Comment by Vaccarini-Lorenzo** [comment]
Hi @liamcain, done!

**Pull request merged**